lighthouse-core/lib/i18n/README.md

```diff
@@ -0,0 +1,305 @@
+         * @example {errorCode} NO_SPEEDLINE_FRAMES
+         */
+        didntCollectScreenshots: `Chrome didn't .... ({errorCode})`,
+    };
+    ```
+
+*   Complex replacement
+
+    When more complex numerical ICU replacement is needed the syntax is mostly
+    the same as direct replacement. This is often used when replcaing times, or
+    percentages.
+
+    Note: these complex ICU formats are automatically given example values based
+    on their ICU format as specified in `collect-strings.js`.  Therefore a normal 
+    description string can be used.
+
+    ```javascript
+    const UIStrings = {
+        displayValueText: 'Interactive at {timeInMs, number, seconds} s',
+    };
+    ```
+
+TODO(exterkamp): explain all the comments and where they go/what they become.
+
+TODO(exterkamp): explain why we can't use some ICU like number formatting.
+
+### The pipeline
+
+The translation pipeline has 2 distinct stages, the collection and translation
+is done pre-compile time, and the replacement is done at runtime.
+
+#### Translation Pipeline (pre-compile)
+
+1.  `file_with_UIStrings.js` this is where strings start. Optimized for
+    programmer ease of use and not for machine parsing. Uses ICU syntax and
+    markdown control characters inline.
+
+2.  `yarn i18n:collect-strings` collects all UIStrings, and generates the
+    pre-locales. Does some parsing to make sure that common mistakes are
+    avoided.
+
+3.  `pre-locale/en-US.json` this is the well formatted _machine parsable_
+    fileset that is uploaded to be translated, i.e. they use $placeholder$
+    syntax instead of ICU. These will never be used by the internal i18n system,
+    they are solely used to send to translators.
+
+4.  `yarn i18n:correct-strings` collects all pre-locales (or returned .json
+    files of all languages if you're a Googler importing strings) and converts
+    them back to Lighthouse json format and puts them into `locales/`.
+
+5.  `locales/{locale}.json` the Lighthouse json files. Used by the i18n.js
+    system to i18n strings. Uses ICU and not $placeholder$ syntax. Optimized for
+    i18n machine use.
+
+This pipeline is best seen with its component yarn commands:
+
+```shell
+# collect UIStrings into pre-locales
+$ yarn i18n:collect-strings
+
+# make the final en-US and en-XL files
+$ yarn i18n:correct-strings
+
+# Send off to translators, and the i18n:correct-strings again
+# once those .json's are done.
+$ sh google_import_script_that_calls_correct_strings
+```
+
+#### String Replacement Pipeline (runtime)
+
+1.  String called in `.js` file, converted to i18n id.
+
+2.  i18n id in lookup table along with backup message.
+
+3.  Message is looked up via `replaceIcuMessageInstanceIds` &
+    `_formatIcuMessage`.
+
+##### Example:
+
+1.  string in `file_with_UIStrings.js`
+
+    ```javascript
+    // Declare UIStrings
+    const UIStrings = {
+      /** Description of a Lighthouse audit that tells the user ...*/
```

Comment:
```suggestion
      /** @description Description of a Lighthouse audit that tells the user ...*/
```

Reply:
This isn't required or suggested yet, so leaving for the follow-up.